src/border_agent/border_agent.cpp

```diff
@@ -311,35 +324,21 @@ void BorderAgent::PublishMeshCopService(void)
     }
 #endif
 
-    mPublisher->PublishService(/* aHostName */ "", otBorderAgentGetUdpPort(instance), kBorderAgentServiceInstanceName,
-                               kBorderAgentServiceType, Mdns::Publisher::SubTypeList{}, txtList);
+    mPublisher->PublishService(/* aHostName */ "", port, kBorderAgentServiceInstanceName, kBorderAgentServiceType,
+                               Mdns::Publisher::SubTypeList{}, txtList);
 }
 
 void BorderAgent::UnpublishMeshCopService(void)
 {
-    assert(IsThreadStarted());
-    VerifyOrExit(!mNetworkName.empty());
-
     otbrLogInfo("Unpublish meshcop service %s.%s.local.", kBorderAgentServiceInstanceName, kBorderAgentServiceType);
 
     mPublisher->UnpublishService(kBorderAgentServiceInstanceName, kBorderAgentServiceType);
```

Comment:
Not related to this PR, just a question: What will happen if the published MeshCoP service has been renamed?

Reply:
First of all. We are not unpublishing MeshCoP services any more. 
For mDNSResponder, the published MeshCoP service will be unpublished, since the implementation stores the original instance name in `mServices`. 
For avahi, we're storing the updated names in `mServices` so that for MeshCoP services, we cannot unpublish the renamed ones. This may be a problem.

Reply:
> First of all. We are not unpublishing MeshCoP services any more.

The same concern applies to case where we want to update the renamed MeshCoP service.

Reply:
Resolved by https://github.com/openthread/ot-br-posix/pull/1104